multiqc/modules/mosaicatcher/mosaicatcher.py

```diff
@@ -0,0 +1,182 @@
+                samples = self._parse_samples(f, samples)
+                self.add_data_source(f)
+
+        self._samples = samples
+
+        self._add_table(samples)
+        self._add_coverage_plot(samples)
+
+        self.write_data_file(self._samples, "multiqc_mosaicatcher")
+        log.info("Found {} samples".format(len(samples)))
+
+    def _add_table(self, samples):
+        self.add_section(
+            plot=table.plot(data=samples, headers=self._setup_headers()),
+        )
+
+    def _add_coverage_plot(self, samples):
+        coverage_plot = bargraph.plot(
+            data=samples,
+            cats=["mapq", "dupl", "good", "reads2", "suppl"],
+            pconfig={
+                "id": "mosaicatcher-coverage",
+                "title": "MosaiCatcher: coverage",
+                "height": 1024,
+                "ymax": 4e6,
+                "ylab": "Cells",
+            },
+        )
+        self.add_section(
+            name="Average coverage per sample",
+            anchor="mosaicatcher-coverage-id",
+            description="Average coverage per sample",
+            plot=coverage_plot,
+        )
+
+    def _setup_headers(self):
+        headers = OrderedDict()
+        for k in list(self._samples.values())[0]:
+            headers[k] = {"title": k, "hidden": True}
+
+        headers["sample"] = {
+            "title": "sample",
+            "description": "Sample (has multiple cells)",
+            "hidden": True,
+        }
+        headers["cell"] = {
+            "title": "cell",
+            "description": "Name of the cell.",
+            "hidden": False,
+        }
+        headers["mapped"] = {
+            "title": "mapped",
+            "max": 4e6,
+            "description": "Total number of reads seen",
+            "scale": "RdYlGn",
+            "hidden": False,
+        }
+        headers["suppl"] = {
+            "title": "suppl",
+            "description": "Supplementary, secondary or QC-failed reads (filtered out)",
+            "hidden": True,
+        }
+        headers["dupl"] = {
+            "title": "dupl",
+            "max": 3e6,
+            "description": "Reads filtered out as PCR duplicates",
+            "scale": "OrRd",
+            "hidden": False,
+        }
+        headers["mapq"] = {
+            "title": "mapq",
+            "description": "Reads filtered out due to low mapping quality",
+            "hidden": True,
+        }
+        headers["read2"] = {
+            "title": "read2",
+            "description": "Reads filtered out as 2nd read of pair",
+            "hidden": True,
+        }
+        headers["good"] = {
+            "title": "good",
+            "max": 8e5,
+            "scale": "RdYlGn",
+            "description": "Reads used for counting.",
+            "hidden": False,
+        }
+        headers["pass1"] = {
+            "title": "pass1",
+            "description": "Enough coverage? If false, ignore all columns from now",
+            "hidden": False,
```

Comment:
Currently it adds a `1.0` value, but since it's a categorical value, perhaps should be a word - e.g. `pass` or `fail`. The column name is also not good, rename `pass1` to something descriptive (`Enough coverage`?)

Reply:
Changed